### PR TITLE
upgrade: suspend and resume instances during the disruptive upgrade

### PR DESCRIPTION
**Pull request by Itxaka**
The services step of the upgrade process wont be able
to pass as long as there are instances running.
We need to stop all instances in case the upgrade method
is disruptive and then resume them afterwards like
a human would do during the disruptive upgrade

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -5302,6 +5321,10 @@ function onadmin_crowbar_nodeupgrade
         fi
 
         if [[ $want_nodesupgrade ]]; then
+            local upgrade_mode=`safely crowbarctl upgrade status --format plain --filter=suggested_upgrade_mode`
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: There's space at the start of this command:
```
# crowbarctl upgrade status --format plain --filter=suggested_upgrade_mode
 non_disruptive
```

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -5302,6 +5321,10 @@ function onadmin_crowbar_nodeupgrade
...
+            if [[ "$upgrade_mode" == "normal" ]]; then
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: What about just `crowbarctl upgrade mode | grep normal` ?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Also, `suggested_upgrade_mode` could be changed by user, then `selected_upgrade_mode` applies. Not happening for mkcloud deployment, but still probably better to just call `crowbarctl upgrade mode`

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4821,6 +4821,25 @@ function oncontroller_waitforinstance
     wait_for 100 1 "ping -q -c 1 -w 1 $vmip >/dev/null" "testvm to boot up"
 }
 
+function oncontroller_suspendallinstances
+{
+    . .openrc
+    for i in `safely openstack server list --all-projects --status active -f value -c ID`; do
```
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Uhm, is `safely` passed to controller?

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: oh wow, nice catch!

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: was expecting to re-use the upgrade_mode var for adding extra steps in the future for the non-disruptive part.

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (Itxaka) [inline scripts/qa_crowbarsetup.sh]: ah yeah, sorry

**Comment by jsuchome** [comment]
You still have that potential inconsistency between suggested and selected upgrade mode

**Comment by Itxaka** [comment]
@jsuchome yep, but as this is only for mkcloud runs, do we need to take into account that part? Are we gonna force a non-disruptive job into a disruptive one?

**Comment by jsuchome** [comment]
It might be possible for mkcloud user to run some parts step by step, and adapting the process manually. I'm doing it, for example. But maybe we do not need to think about it for automated deployment.

However, it seems easy to just do something like 
```
mode=normal
if crowbarctl upgrade mode | grep non_disruptive ; then
    mode=non_disruptive
fi
```

Sorry for syntax, did not check it myself...

**Comment by Itxaka** [comment]
ahh indeed @jsuchome , changed plus added a couple of comments in there

**Pull request merged**